crates/rethnet_evm/benches/state_history.rs

```diff
@@ -0,0 +1,100 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rethnet_eth::B256;
+
+mod util;
+use util::{bench_sync_state_method, prep_no_op, SNAPSHOT_SCALES};
+
+fn bench_checkpoint(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateHistory::checkpoint()",
+        prep_no_op,
+        |mut state, _number_of_accounts| {
+            let result = state.checkpoint();
+            debug_assert!(result.is_ok());
+        },
+        &[0],
+        &[0],
+    );
+}
+
+fn bench_make_snapshot(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateHistory::make_snapshot",
+        prep_no_op,
+        |mut state, _number_of_accounts| {
+            state.make_snapshot();
+        },
+        &[0],
+        &SNAPSHOT_SCALES,
```

Comment:
The number of snapshots doesn't matter. No need to vary this for snapshot creation. The thing that would be of interest is how the performance scales depending on the other 3 variables:
- number of accounts
- number of checkpoints
- number of storage slots per account

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3921/commits/797e797c542cda249f9e0bac462d0649ebb27cf9